app/reactions/server/setReaction.js

```diff
@@ -9,8 +9,9 @@ import { isTheLastMessage, msgStream } from '../../lib';
 import { hasPermission } from '../../authorization/server/functions/hasPermission';
 import { api } from '../../../server/sdk/api';
 
-const removeUserReaction = (message, reaction, username) => {
+const removeUserReaction = (message, reaction, username, uid) => {
 	message.reactions[reaction].usernames.splice(message.reactions[reaction].usernames.indexOf(username), 1);
+	message.reactions[reaction].uids.splice(message.reactions[reaction].uids.indexOf(uid), 1);
```

Comment:
@yash-rajpal 
There exist older messages which do not have the `uids` property in their reactions
This might throw an error when the user tries to remove his *old* reaction

Reply:
Yeah, Good point. I will take care of this.

Thanks for mentioning.